### DISCUSSION TITLE
Let `showBZDemo` generate BCS style Brillouin zone plot.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [goodluck1982/SpaceGroupIrep#31] [General]
[Bilbao Crystallographic Server](https://www.cryst.ehu.es/) also provides the Brillouin zone plot stored in the PostScript file format, for example, [this one](https://www.cryst.ehu.es/cryst/kvec/fig/f6mmP.ps):

![image](https://user-images.githubusercontent.com/11155854/153694959-90ec27e2-1a0b-489a-943c-62a6bc2cad67.png)

The following is the similar plot generated by the `showBZDemo` function in `SpaceGroupIrep`:

![image](https://user-images.githubusercontent.com/11155854/153695156-2d9c03e5-795a-4ddb-abd0-7d47fe28a46f.png)

It would be better if `showBZDemo` can also generate BCS style Brillouin zone plot.

BTW, what's your IDE for [Wolfram Language](https://www.wolfram.com/language) development? I've noticed the following tools/plugins for this purpose:

1. [Wolfram Workbench](https://www.wolfram.com/workbench/) is the official Eclipse-based IDE for the Wolfram Language provided by Wolfram, and is free.
2. [Wolfram Language](https://plugins.jetbrains.com/plugin/7232-wolfram-language) is a plugin of JetBrains IntelliJ IDEA, but not free.

Regards,
HZ

**Comment by goodluck1982** [comment]
As explained by its name, `showBZDemo` only shows the BZ as a demonstration. Its purpose is to give a 3D rotatable  visulization, facilitating to watch the positions of high-symmetry k points. A 2D static picture is not convenient for this purpose.

As for static 2D pictures, one can simply refer to the figures in the BC book, such as 
![图片](https://user-images.githubusercontent.com/48718961/153703450-45c4c07f-2803-4b10-a65c-b6fd4a6316be.png)

**Comment by goodluck1982** [comment]
> BTW, what's your IDE for [Wolfram Language](https://www.wolfram.com/language) development? I've noticed the following tools/plugins for this purpose:
> 
>     1. [Wolfram Workbench](https://www.wolfram.com/workbench/) is the official Eclipse-based IDE for the Wolfram Language provided by Wolfram, and is free.
> 
>     2. [Wolfram Language](https://plugins.jetbrains.com/plugin/7232-wolfram-language) is a plugin of JetBrains IntelliJ IDEA, but not free.

Thanks for your information.
I just use mathematica for most of the time, and sometimes I also use [VS code](https://code.visualstudio.com/).

**Comment by hongyi-zhao** [comment]
I tried to open [`SpaceGroupIrep.wl`](https://github.com/goodluck1982/SpaceGroupIrep/blob/main/SpaceGroupIrep.wl)  with [Wolfram Workbench](https://www.wolfram.com/workbench/). The effect looks very good, as shown in the screenshot below:

![image](https://user-images.githubusercontent.com/11155854/153705345-8da8d45f-0c4e-45fe-968f-f81e687ca149.png)

![image](https://user-images.githubusercontent.com/11155854/153710069-f0952347-813c-4211-bc4d-82f7b6234013.png)


As for [Wolfram Language](https://plugins.jetbrains.com/plugin/7232-wolfram-language), I haven't tried it because it requires users to apply for a license before installation and use. But according to the presentation on its official plug-in site, it's also great.
